internal/controller/vrg_recipe.go

```diff
@@ -330,6 +338,100 @@ func recipeNamespacesValidate(recipeElements util.RecipeElements, vrg ramen.Volu
 	return nil
 }
 
+// If VM protection is requested, Velero and the kubevirt plugin must be present,
+// unless an alternate supported solution is configured.
+func recipeVMBackupValidate(ctx context.Context, reader client.Reader, recipeElements util.RecipeElements,
+	vrg ramen.VolumeReplicationGroup,
+	ramenConfig ramen.RamenConfig, log logr.Logger,
+) error {
+	vmProtection := isProtectingVirtualMachine(recipeElements, vrg, log)
+
+	if !vmProtection {
+		return nil
+	}
+
+	if ramenConfig.KubeObjectProtection.VeleroNamespaceName != "" ||
+		len(ramenConfig.KubeObjectProtection.VeleroNamespaceName) > 0 {
+		if isKubeVirtEnabled(ctx, reader, ramenConfig, log) {
+			return nil
+		}
+
+		return fmt.Errorf("kubevirt plugin is disabled, hence VM resources backup/restore might not succeed")
+	}
+
+	log.Info("Skipping kubevirt-plugin validation as Velero is not configured;" +
+		"VM protection requires Velero and the kubevirt plugin unless an alternate supported solution is in place.")
+
+	return nil
+}
+
+func isProtectingVirtualMachine(recipeElements util.RecipeElements, vrg ramen.VolumeReplicationGroup,
+	log logr.Logger,
+) bool {
+	vmResourceTypes := map[string]bool{
+		"virtualmachine":             true,
+		"virtualmachine.kubevirt.io": true,
+	}
+	if vrg.Spec.KubeObjectProtection.RecipeRef.Name == recipecore.VMRecipeName {
+		log.Info("Recipe protects VM resources")
+
+		return true
+	}
+
+	// ToDO: Add validatation on recipe protecting VMs without using inbuilt VM-recipe
+	for _, group := range recipeElements.RecipeWithParams.Spec.Groups {
+		if len(group.IncludedResourceTypes) > 0 {
+			for _, resourceTypes := range group.IncludedResourceTypes {
+				if vmResourceTypes[strings.ToLower(strings.TrimSpace(resourceTypes))] {
+					log.Info("Recipe protects VM resources")
+
+					return true
+				}
+			}
+		}
+	}
+
+	log.Info(fmt.Sprintf("%s recipe is not protecting VM resource", recipeElements.RecipeWithParams.Name))
+
+	return false
+}
+
+func isKubeVirtEnabled(ctx context.Context, reader client.Reader,
+	ramenConfig ramen.RamenConfig, log logr.Logger,
+) bool {
+	log.Info("Checking if kubevirt is enabled")
+	// Read Velero deployment and check if kubevirt plugin is enabled
+	var veleroDeploy appsv1.Deployment
+
+	veleroLookUpKey := types.NamespacedName{
+		Name:      veleroDeployment,
+		Namespace: ramenConfig.KubeObjectProtection.VeleroNamespaceName,
+	}
+	// Fetch the Velero deployment
+	if err := reader.Get(ctx, veleroLookUpKey, &veleroDeploy); err != nil {
+		log.Error(err, "failed to get Velero deployment")
+
+		return false
+	}
+
+	var initContainerNames []string
+	// Go through InitContainers to check if kubevirt plugin is enabled
+	for _, ic := range veleroDeploy.Spec.Template.Spec.InitContainers {
+		if strings.Contains(ic.Name, kubevirtVeleroPluginName) {
+			log.Info("kubevirt is enabled")
+
+			return true
+		}
+		initContainerNames = append(initContainerNames, ic.Name)
+	}
```

Comment:
Here it's better to avoid copying by iterating by index:
```suggestion
initContainers := veleroDeploy.Spec.Template.Spec.InitContainers
initContainerNames := make([]string, 0, len(initContainers))

// Go through InitContainers to check if kubevirt plugin is enabled
for i := range initContainers {
	if strings.Contains(initContainers[i].Name, kubevirtVeleroPluginName) {
		log.Info("kubevirt is enabled")
		return true
	}
	initContainerNames = append(initContainerNames, initContainers[i].Name)
}

```